lib/compiler/aro_translate_c.zig

```diff
@@ -496,16 +711,126 @@ fn transType(c: *Context, scope: *Scope, raw_ty: Type, source_loc: TokenIndex) T
         .long_double => return ZigTag.type.create(c.arena, "c_longdouble"),
         .float80 => return ZigTag.type.create(c.arena, "f80"),
         .float128 => return ZigTag.type.create(c.arena, "f128"),
+        .@"enum" => @panic("TODO"),
+        .pointer,
+        .unspecified_variable_len_array,
+        .array,
+        .static_array,
+        .incomplete_array,
+        => @panic("TODO"),
         .func,
         .var_args_func,
         .old_style_func,
-        => return transFnType(c, scope, raw_ty, ty, source_loc, .{}),
+        => return transFnType(c, scope, ty, ty, source_loc, .{}),
+        .@"struct",
+        .@"union",
+        => {
+            var trans_scope = scope;
+            if (ty.isAnonymousRecord(c.comp)) {
+                const record_decl = ty.data.record;
+                const name_id = c.mapper.lookup(record_decl.name);
+                if (c.weak_global_names.contains(name_id)) trans_scope = &c.global_scope.base;
+            }
+            const name = c.decl_table.get(@intFromPtr(ty.data.record)).?;
+            return ZigTag.identifier.create(c.arena, name);
+        },
+        .attributed,
+        .typeof_type,
+        .typeof_expr,
+        => unreachable,
         else => return error.UnsupportedType,
     }
 }
 
-fn zigAlignment(bit_alignment: u29) u32 {
-    return bit_alignment / 8;
+// This function returns an array of ?c_uint to match Clang's behaviour of using c_uint.
```

Comment:
This comment is out of date. Also let's make it a doc comment (`///` instead of `//`)